src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -146,10 +146,53 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		return this._attachmentModel;
 	}
 
-	public getAttachedAndImplicitContext(): IChatRequestVariableEntry[] {
+	public getAttachedAndImplicitContext(
+		chatWidget: IChatWidget,
+	): IChatRequestVariableEntry[] {
 		const contextArr = [...this.attachmentModel.attachments];
-		if (this.implicitContext?.enabled && this.implicitContext.value) {
-			contextArr.push(this.implicitContext.toBaseEntry());
+
+		if (this._implicitContext?.enabled && this._implicitContext.value) {
+			const mainEntry = this._implicitContext.toBaseEntry();
+
+			contextArr.push(mainEntry);
+
+			// if the implicit context is a file, it can have nested
+			// file references that should be included in the context
+			if (this._implicitContext.validFileReferenceUris) {
+				const childReferences = this._implicitContext.validFileReferenceUris
+					.map((uri): IBaseChatRequestVariableEntry => {
+						return {
+							...mainEntry,
+							name: `file:${basename(uri.path)}`,
+							value: uri,
+						};
+					});
+				contextArr.push(...childReferences);
+			}
+		}
+
+		// factor in nested references of dynamic variables into the implicit attached context
+		for (const part of chatWidget.parsedInput.parts) {
+			if (!(part instanceof ChatRequestDynamicVariablePart)) {
+				continue;
+			}
+
+			if (!(part.isFile && URI.isUri(part.data))) {
+				continue;
+			}
+
+			for (const childUri of part.childReferences ?? []) {
+				contextArr.push({
+					id: part.id,
+					name: basename(childUri.path),
+					value: childUri,
+					kind: 'implicit',
```

Comment:
It's not 'implicit' context if it's referenced in the prompt

Reply:
The `root` variable is referenced in the prompt, not the nested file references, hence the thinking was that the child references are "implicitly" referenced. What category you'd use instead here?

Reply:
I would say that the child references are just part of the root reference, and they should be the same type.

Reply:
Then we should move these somewhere else? The `IChatRequestVariableEntry` does not currently have the "dynamic" kind.